hilti/runtime/src/tests/map.cc

```diff
@@ -33,40 +33,12 @@ TEST_CASE("subscript") {
     SUBCASE("mut lvalue") {
         Map<int, int> m;
 
-        m[1] = 11;
+        m.index_assign(1, 11);
         CHECK(m.contains(1));
-        CHECK_EQ(m[1].operator int(), 11);
+        CHECK_EQ(m[1], 11);
         int m1 = m[1];
         CHECK_EQ(m1, 11);
-        CHECK_THROWS_WITH_AS(m[99].operator int(), "key is unset", const IndexError&);
-
-        // Proxy objects perform lazy insertion.
-        auto p = m[2];
-        (void)p;
-        CHECK(! m.contains(2));
-
-        // Proxy objects stringify to the referenced value if it exists, and throw otherwise.
-        CHECK_EQ(to_string(m[1]), "11");
-        CHECK_THROWS_WITH_AS(to_string(m[2]), "key is unset", const IndexError&);
-
-        CHECK_EQ(fmt("%s", m[1]), "11");
-
-        // Proxy objects only invalidate iterators if an element was actually inserted.
```

Comment:
Should we keep this? Looks like we still have the same iterator invalidation semantics?